Plans Step: fix design alignment issues

mtias opened the pull request:
The current plan design has issues in different locales, where the plan price and button actions don't align well and create a sloppy stair effect. This adds a flex wrapper and ensure the bottom section is aligned.

**Before:**
<img width="1051" alt="Screenshot 2022-07-02 at 11 10 26" src="https://user-images.githubusercontent.com/548849/177124581-f40952d3-86ba-423e-8ecc-b6abc7b2981d.png">

**After:**
<img width="1192" alt="Screenshot 2022-07-04 at 11 17 40" src="https://user-images.githubusercontent.com/548849/177124606-2ddc77d6-dae7-4fc5-920e-fce3c1bcde10.png">

mtias committed:
Plans Step: fix design alignment in various locales.

Use a flex wrapper to ensure the buttons are aligned at the bottom

gmovr reviewed (approved):
Thanks for sorting this! These changes look good to me. Tested in ES and EN. 🚢

gmovr reviewed (changes requested):
I tested initially in Safari and Chrome on mobile/desktop. But when testing in Firefox on large viewports, it overlaps the table in the signup flow.
<img width="1133" alt="Screenshot 2022-07-04 at 12 42 37" src="https://user-images.githubusercontent.com/52675688/177140119-eae04bfe-4332-405f-af57-db83a92b9d4c.png">
<img width="1338" alt="Screenshot 2022-07-04 at 12 44 43" src="https://user-images.githubusercontent.com/52675688/177140129-21560dfa-4e7a-4656-9940-c0c842784e6c.png">
<img width="1312" alt="Screenshot 2022-07-04 at 12 48 49" src="https://user-images.githubusercontent.com/52675688/177140133-16cb1e0c-6306-49a5-b6d2-d749dee25c34.png">

ciprianimike committed:
Update/plan grid flex design extra row (#65302)

* add extra <th> row in price comparison table

Split PlansComparisonColHeader contents with the new component PlansComparisonColCTA and move them to the new extra <th> row

* Comparison table header CSS tuning

* CSS tuning

Co-authored-by: Michele Cipriani <michele@MichelesMBP2022.station>

chriskmnds commented:
Rebased and cleaned up some conflicts. Let's do some testing to make sure nothing was done wrong there 😬 cc @gmovr 

Also, I see some errors in the console concerning the use of `:first-child`/`:nth-child` in the styles. @ciprianimike can you maybe take a look? 🙏 

e.g. 
<img width="760" alt="Screenshot 2022-07-14 at 2 41 49 PM" src="https://user-images.githubusercontent.com/1705499/178974595-3bad26d4-8510-4856-a121-d07950a3bfbd.png">

gmovr commented:
Awesome! I'll take it for a spin with the experiment from https://github.com/Automattic/wp-calypso/pull/65126 as well 👍

chriskmnds committed:
hide 24-hr support from highlights on monthly

chriskmnds committed:
refactor first/nth-child -> first/nth-of-type

gmovr reviewed (approved):
Changes look good to me! Tested in Firefox and Chrome again in various viewports. 🚀
